test/CrowdsaleGenTest.js

```diff
@@ -154,12 +158,30 @@ contract('LifCrowdsale Property-based test', function(accounts) {
       state.status = 3;
     } catch (e) {
       if (!shouldThrow)
-        throw(new ExceptionRunningCommand(e, state, command));   
+        throw(new ExceptionRunningCommand(e, state, command));
     }
 
     return state;
   };
 
+  let runFinalizeCrowdsaleCommand = async (command, state) => {
+    let shouldThrow = (state.crowdsalePaused == command.pause) ||
+      (command.fromAccount != 0) ||
+      !state.crowdsaleContract.isFinalized ||
+      !(await state.crowdsaleContract.hasEnded());
```

Comment:
we should do this kind of checks on the on-memory state, not on the contract. So if `hasEnded` is not in the on-memory state yet, we should add it